src/go/k8s/go.mod

```diff
@@ -25,41 +25,55 @@ require (
 	github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring v0.74.0
 	github.com/prometheus/client_golang v1.20.0
 	github.com/prometheus/common v0.55.0
-	github.com/redpanda-data/common-go/rpadmin v0.1.5-0.20240814205445-8e6eb5806561
+	github.com/redpanda-data/common-go/rpadmin v0.1.7-0.20240909210234-44a1212753d7
 	github.com/redpanda-data/console/backend v0.0.0-20240303221210-05d5d9e85f20
 	github.com/redpanda-data/helm-charts v0.0.0-20240911060052-2bf9dd6f0996
-	github.com/redpanda-data/redpanda/src/go/rpk v0.0.0-20240105044330-c094966ca0cf
+	github.com/redpanda-data/redpanda/src/go/rpk v0.0.0-20240827155712-244863ea0ae8
+	github.com/spf13/afero v1.11.0
 	github.com/spf13/cobra v1.8.1
 	github.com/stretchr/testify v1.9.0
+	github.com/testcontainers/testcontainers-go v0.32.0
 	github.com/testcontainers/testcontainers-go/modules/redpanda v0.32.0
-	github.com/twmb/franz-go v1.16.1
-	github.com/twmb/franz-go/pkg/kadm v1.11.0
-	github.com/twmb/franz-go/pkg/kmsg v1.7.0
+	github.com/twmb/franz-go v1.17.0
+	github.com/twmb/franz-go/pkg/kadm v1.12.0
+	github.com/twmb/franz-go/pkg/kmsg v1.8.0
 	github.com/twmb/franz-go/pkg/sasl/kerberos v1.1.0
 	go.uber.org/zap v1.27.0
 	gopkg.in/yaml.v2 v2.4.0
 	gopkg.in/yaml.v3 v3.0.1
 	helm.sh/helm/v3 v3.14.4
-	k8s.io/api v0.29.5
-	k8s.io/apiextensions-apiserver v0.29.5
-	k8s.io/apimachinery v0.29.5
-	k8s.io/client-go v0.29.5
+	k8s.io/api v0.30.3
+	k8s.io/apiextensions-apiserver v0.30.1
+	k8s.io/apimachinery v0.30.3
+	k8s.io/client-go v0.30.3
 	k8s.io/component-helpers v0.29.5
 	k8s.io/kubectl v0.29.5
 	k8s.io/utils v0.0.0-20240711033017-18e509b52bc8
 	pgregory.net/rapid v1.1.0
-	sigs.k8s.io/controller-runtime v0.17.2
+	sigs.k8s.io/controller-runtime v0.18.5
```

Comment:
Not entirely sure on all of these changes, but I know that a bunch of Kubernetes stuff along with `controller-runtime` are _**highly**_ sensitive to minor version changes in the underlying libraries. Just looking above it looks like we also have `kubectl` and `component-helpers` at v0.29.5... I'm not sure if that'll cause odd problems or not.

I think that the `helm-charts` repo also has a `controller-runtime` dependency? Might version mismatches there cause problems here?

Reply:
I've yet to experience any issues from version mismatches that don't appear at either compile time or very early on in startup / `init()` code. Have you seen runtime issues before?

> I think that the helm-charts repo also has a controller-runtime dependency? Might version mismatches there cause problems here?

It does but it uses very little of the API(s) exposed. I'm fairly certain that it's just the `client.Client` and some configuration helpers. I don't imagine any issues occurring?

Reply:
Synced all the `k8s.io/*` deps with the exception of `k8s.io/utils` which doesn't really have any published versions.